File: leetcode-js/228-summary-ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {
    if (nums.length === 1) {
        return [`${nums[0]}`];
    }

    const result = [];

    for (let i = 0; i < nums.length; i += 1) {
        const n = nums[i];
        while (i + 1 < nums.length && nums[i+1] - nums[i] === 1) {
            i += 1;
        }

        if (nums[i] !== n) {
            result.push(`${n}->${nums[i]}`);
        } else {
            result.push(`${n}`);
        }
    }

    return result;
};

File: leetcode-java/228-summary-ranges.java
public class Solution {
    public List<String> summaryRanges(int[] nums) {
        if (nums.length == 1) {
            return Arrays.asList(Integer.toString(nums[0]));
        }

        List<String> res = new ArrayList<>();

        int i = 0;

        while (i < nums.length) {
            int a = nums[i];
            while (i + 1 < nums.length && nums[i + 1] - nums[i] == 1) {
                i++;
            }
            if (a == nums[i]) {
                res.add(Integer.toString(a));
            } else {
                res.add(a + "->" + nums[i]);
            }
            i++;
        }

        return res;
    }
}
